src/plugins/intel_gpu/src/kernel_selector/kernels/gemm/gemm_kernel_tiled_opt.cpp

```diff
@@ -53,25 +55,34 @@ GemmKernelTiledOpt::GemmTuningData GemmKernelTiledOpt::SetTuningParams(const gem
 
     GemmKernelTiledOpt::GemmTuningData tuning_data;
 
-    auto m_size = output.Y().v;
-    auto n_size = output.X().v;
-    auto k_size = params.transpose_input0 ? params.inputs[0].Y().v : params.inputs[0].X().v;
+     if (!params.has_dynamic_tensors()) {
+        auto m_size = output.Y().v;
```

Comment:
During network execution, after update_shape() function call, params will not contain dynamic tensors, so even for shape agnostic dynamic kernel tuning will be performed in this if..else branch. Tuning function may change tiles sizes which will lead to incorrect workgroup sizes configuration in SetDefault() function after update_dispatch_data_func() call. It looks like you need to have some information if SetTuningParams call performed for dynamic kernel or fully static, and if it is dynamic case then always return tiles sizes aligned with simd_size=16, isn't it?

Reply:
Yeah, we can add a common kernel param of is_shape_agnostic ?

Reply:
Updated to add `is_shape_agnostic` to `base_params`